### DISCUSSION TITLE
Shell script for easing transition from 'paster' to 'ckan'

### DISCUSSION THREAD

**Discussion by ThrawnCA** [ckan/ckan#6039] [Show and tell]
If anyone's interested, we've devised a shell script for making the transition from `paster` easier, at https://github.com/qld-gov-au/opswx-ckan-cookbook/blob/master/files/default/ckan_cli

The intent is that you can place it in your virtualenv `bin` directory and then call it with the same syntax for CKAN 2.8 or 2.9, eg:

    ckan_cli search-index rebuild
    ckan_cli user list

It checks for the presence of paster or ckan, using whichever one it finds, and handles the argument order for you.

Config file location defaults to `/etc/ckan/default/production.ini` (since that's the old default, and if you're migrating, your config file is probably still there), but can be overridden by setting the `CKAN_INI_FILE` environment variable.

**Comment by amercader** [comment]
That looks useful @ThrawnCA! The convention in CKAN is to use [`CKAN_INI`](https://docs.ckan.org/en/2.9/maintaining/cli.html) as the env var name in case you want to align with it.

**Reply by ThrawnCA** [reply]
Thanks for the feedback. I've updated it when copying the script to our other repos, and will likely update our cookbook next time I'm working on it.

**Comment by ThrawnCA** [comment]
An updated version is at https://github.com/qld-gov-au/ckan-ex-qgov/blob/develop/.docker/scripts/ckan_cli to handle some differences in syntax.

- `paster` uses 'create-test-data', `ckan` uses 'seed'
- `ckan` uses 'user show <name>', `paster` just uses 'user <name>'
